app/js/spatial/geo.js

```diff
@@ -95,6 +97,56 @@ export function watchLocation(callback) {
   );
 }
 
+/*
+  I think we should probably have this bit of mapping enumerateTilesAround() be executed only once 
+  and passed to different functions, because as it is now, this first part is done twice for both this function
+  and the announcer function to get all the tiles features.
+  - Kris
+*/
+// Gets the street from nearest address and returns its name and city
+export async function getCurrentRoad(locationProvider){
+  const features = await Promise.all(
+    enumerateTilesAround(locationProvider.latitude, locationProvider.longitude, locationProvider.radiusMeters)
+    .map(t => {
+      t.load();
+      return t.getFeatures();
+    })
+  )
+  .then(tileFeatures => {
+    const reduced = tileFeatures
+      // Flatten list of features across all nearby tiles
+      .reduce((acc, cur) => acc.concat(cur), [])
+      // Limit to roads
+      .filter(
+        f => f.feature_type == "highway" && 
+        f.geometry.type == "LineString" &&
+        f.feature_value == "primary" ||
+        f.feature_value == "residential" ||
+        f.feature_value == "tertiary" 
```

Comment:
I think you're missing some parentheses here: you want the first two conditions and one of the final three to be true, but this will return true if just either of the final two are true. (This might accidentally work in practice, since it's possible only `highway` `LineString` features can have the specified values, anyway.) The following is better:
```suggestion
        f => f.feature_type == "highway" && 
        f.geometry.type == "LineString" &&
        f.feature_value in ["primary", "residential", "tertiary"] 
```